utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import android.os.Bundle
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.extensions.getProto
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProto
+import org.oppia.android.util.extensions.putProtoExtra
+
+/** Represents key for Profile ID. */
+const val PROFILE_ID_INTENT_DECORATOR = "CurrentUserProfileIdIntentDecorator.profile_id_key"
+private const val PROFILE_ID_BUNDLE_DECORATOR =
+  "CurrentUserProfileIdIntentDecorator.profile_id_bundle_key"
+
+/** Decorator that allows an activity to wrap a user's [ProfileId] within its intent. */
+object CurrentUserProfileIdIntentDecorator {
```

Comment:
Per my comment in the proto file, is this actually needed if we're including ProfileId directly in the params & arguments used to create activities & fragments?

Reply:
Yes. The main reason for introducing **CurrentUserProfileIdIntentDecorator**  in this PR is to support [NavigationDrawerFragmentPresenter ](https://github.com/oppia/oppia-android/blob/3a369ed0994395dde2884331f50a8edb0c8db7aa/app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt#L72)receive profileId from HomeActivity, HelpActivity, OptionActivity, AdministratorControlsActivity, DeveloperOptionsActivity using same key.